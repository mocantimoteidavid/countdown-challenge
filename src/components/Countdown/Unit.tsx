import styled from "styled-components"

interface Props {
  label: string
  value: number
  leadingZero?: boolean
}

const Label = styled.label`
  text-transform: uppercase;
  font-size: 10px;
  margin: 0;
  padding: 0;
`

const Value = styled.div`
  font-size: 38px;
  line-height: 45px;
  font-weight: 600;
  font-feature-settings: "tnum";
  font-variant-numeric: tabular-nums;
`

const Wrapper = styled.div`
  border: 1px solid #040404;
  padding: 1px 4px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-width: 50px;
  height: 58px;
`

const Unit: React.FC<Props> = ({ label, value, leadingZero }) => {
  const isValueSingleDigit = value < 10 && value >= 0
  const displayValue = isValueSingleDigit && leadingZero ? `0${value}` : value

  return (
    <Wrapper>
      <Value data-testid="value">{displayValue}</Value>
      <Label data-testid="label">{label}</Label>
    </Wrapper>
  )
}

export default Unit
