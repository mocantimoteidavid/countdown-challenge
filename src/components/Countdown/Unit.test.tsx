import { render, screen } from "@testing-library/react"
import Unit from "./Unit"

describe("Unit", () => {
  it("renders label and value correctly", () => {
    render(<Unit label="Pants" value={26} />)

    const renderedValue = screen.getByTestId("value").innerHTML
    const renderedLabel = screen.getByTestId("label").innerHTML

    expect(renderedValue).toEqual("26")
    expect(renderedLabel).toEqual("Pants")
  })

  it("renders leading 0 for one digit value when leadingZero is true", () => {
    render(<Unit label="Pants" value={7} leadingZero />)

    const renderedValue = screen.getByTestId("value").innerHTML

    expect(renderedValue).toEqual("07")
  })

  it("does not render leading 0 for two digits value when leadingZero is true", () => {
    render(<Unit label="Pants" value={20} leadingZero />)

    const renderedValue = screen.getByTestId("value").innerHTML

    expect(renderedValue).toEqual("20")
  })

  it("does not render leading 0 for one digit value when leadingZero is false", () => {
    render(<Unit label="Pants" value={7} />)

    const renderedValue = screen.getByTestId("value").innerHTML

    expect(renderedValue).toEqual("7")
  })

  it("renders with correct styling", () => {
    const { container } = render(<Unit label="Pants" value={7} />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
