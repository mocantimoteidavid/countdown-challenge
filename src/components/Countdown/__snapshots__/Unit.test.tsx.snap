// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit renders with correct styling 1`] = `
.c2 {
  text-transform: uppercase;
  font-size: 10px;
  margin: 0;
  padding: 0;
}

.c1 {
  font-size: 38px;
  line-height: 45px;
  font-weight: 600;
  font-feature-settings: "tnum";
  font-variant-numeric: tabular-nums;
}

.c0 {
  border: 1px solid #040404;
  padding: 1px 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 50px;
  height: 58px;
}

<div
  class="c0"
  data-testid="unit"
>
  <div
    class="c1"
    data-testid="value"
  >
    7
  </div>
  <label
    class="c2"
    data-testid="label"
  >
    Pants
  </label>
</div>
`;
